tests/FSharp.Compiler.ComponentTests/Language/AttributeCheckingTests.fs

```diff
@@ -41,4 +41,54 @@ type C() =
         """
          |> ignoreWarnings
          |> compile
-         |> shouldSucceed
\ No newline at end of file
+         |> shouldSucceed
+
+    [<Fact>]
+    let ``ObsoleteAttribute is not taken into account when used on a type.`` () =
```

Comment:
Is this correct? are we supposed to ignore obsolete attribute on a type? If so that seems a bit wierd.

Reply:
No that was a mistake from my part. I was trying to make the test pass without making sure the test was valid.  my bad .

Reply:
Updated the PR description with the testing scenarios